spec/jobs/inherit_permissions_job_spec.rb

```diff
@@ -2,92 +2,199 @@
+      let(:access) { 'edit' }
+
+      it 'copies permissions to its contained files' do
+        # files have the depositor as the edit user to begin with
+        expect(work.file_sets.first.edit_users).to eq [user.to_s]
+
+        described_class.perform_now(work, use_valkyrie: false)
+        work.reload.file_sets.each do |file|
+          expect(file.edit_users).to match_array [user.to_s, "abc@123.com"]
+        end
+      end
 
-  context "when edit people change" do
-    let(:name) { 'abc@123.com' }
-    let(:type) { 'person' }
-    let(:access) { 'edit' }
+      context "when people should be removed" do
+        before do
+          file_set = work.file_sets.first
+          file_set.permissions.build(name: "remove_me", type: type, access: access)
+          file_set.save
+        end
 
-    it 'copies permissions to its contained files' do
-      # files have the depositor as the edit user to begin with
-      expect(work.file_sets.first.edit_users).to eq [user.to_s]
+        it 'copies permissions to its contained files' do
+          # files have the depositor as the edit user to begin with
+          expect(work.file_sets.first.edit_users).to eq [user.to_s, "remove_me"]
 
-      described_class.perform_now(work)
-      work.reload.file_sets.each do |file|
-        expect(file.edit_users).to match_array [user.to_s, "abc@123.com"]
+          described_class.perform_now(work, use_valkyrie: false)
+          work.reload.file_sets.each do |file|
+            expect(file.edit_users).to match_array [user.to_s, "abc@123.com"]
+          end
+        end
       end
     end
 
-    context "when people should be removed" do
-      before do
-        file_set = work.file_sets.first
-        file_set.permissions.build(name: "remove_me", type: type, access: access)
-        file_set.save
+    context "when read people change" do
+      let(:name) { 'abc@123.com' }
+      let(:type) { 'person' }
+      let(:access) { 'read' }
+
+      it 'copies permissions to its contained files' do
+        # files have the depositor as the edit user to begin with
+        expect(work.file_sets.first.read_users).to eq []
+
+        described_class.perform_now(work, use_valkyrie: false)
+        work.reload.file_sets.each do |file|
+          expect(file.read_users).to match_array ["abc@123.com"]
+          expect(file.edit_users).to match_array [user.to_s]
+        end
       end
+    end
+
+    context "when read groups change" do
+      let(:name) { 'my_read_group' }
+      let(:type) { 'group' }
+      let(:access) { 'read' }
 
       it 'copies permissions to its contained files' do
         # files have the depositor as the edit user to begin with
-        expect(work.file_sets.first.edit_users).to eq [user.to_s, "remove_me"]
+        expect(work.file_sets.first.read_groups).to eq []
 
-        described_class.perform_now(work)
+        described_class.perform_now(work, use_valkyrie: false)
         work.reload.file_sets.each do |file|
```

Comment:
In reading this spec, it would be possible that we have an empty array, which would mean the spec is not testing anything.

It may be helpful to capture the `Hyrax.query_service.custom_queries.find_child_filesets(resource: resource)` as a variable and say that you expect it to have a specific count/size.